### PR TITLE
Add limited Windows support

### PR DESCRIPTION
**Pull request by drew-richardson**

### CODE REVIEW

File: platform_windows.cs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+        public void Start()
+        {
+            Application.EnableVisualStyles();
+            Application.SetCompatibleTextRenderingDefault(false);
+            rootWindow = new RootWindow(checkQuit);
+            willFinishStartup();
+            didFinishStartup();
+            // DR Is this the right place to call these?
+            willActivate();
```
Comment (richardwilkes) [inline platform_windows.cs:67]: This is intended to let us know when the app is moved to the front for user interaction, usually due to one of its windows being clicked on. It should only be emitted when the front/current app changes, not for every click on a window.

File: platform_windows.cs [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+            this.quitAfterLastWindowClosed = Marshal.GetDelegateForFunctionPointer<BoolCallback>(quitAfterLastWindowClosed);
+            this.checkQuit = Marshal.GetDelegateForFunctionPointer<Int32Callback>(checkQuit);
+        }
+
+        public void Start()
+        {
+            Application.EnableVisualStyles();
+            Application.SetCompatibleTextRenderingDefault(false);
+            rootWindow = new RootWindow(checkQuit);
+            willFinishStartup();
```
Comment (richardwilkes) [inline platform_windows.cs:64]: Once this is called, it should be safe for the app developer to make calls to the library to create windows and other wise interact with things. Is that the case here?

File: platform_windows.cs [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+            this.didDeactivate = Marshal.GetDelegateForFunctionPointer<VoidCallback>(didDeactivate);
+            this.quitAfterLastWindowClosed = Marshal.GetDelegateForFunctionPointer<BoolCallback>(quitAfterLastWindowClosed);
+            this.checkQuit = Marshal.GetDelegateForFunctionPointer<Int32Callback>(checkQuit);
+        }
+
+        public void Start()
+        {
+            Application.EnableVisualStyles();
+            Application.SetCompatibleTextRenderingDefault(false);
+            rootWindow = new RootWindow(checkQuit);
```
Comment (richardwilkes) [inline platform_windows.cs:63]: rootWindow? Does this encompass all windows the user might create?

File: platform_windows.go [hunk 4]
```diff
@@ -30,19 +79,33 @@ func platformInvokeAfter(id uint64, after time.Duration) {
 
 // ----- Menu section -----
 
+var (
+	appBar *MenuBar
```
Comment (richardwilkes) [inline platform_windows.go:83]: I didn't think Windows had a concept of a global menu bar? Doesn't this need to be a per-window thing?

File: build.bat [hunk 5]
```diff
@@ -0,0 +1,14 @@
+@echo off
+
```
Comment (richardwilkes) [inline build.bat]: Let's rename this to something like `build_windows.bat` to make it clear

**Review by richardwilkes — CHANGES_REQUESTED** [review]

File: platform_windows.cs (hunk 3 above)
Comment (drew-richardson) [reply platform_windows.cs:63]: No. They are visually distinct windows (ie, not ugly sub windows of some outer window).

File: platform_windows.cs (hunk 2 above)
Comment (drew-richardson) [reply platform_windows.cs:64]: Yes, the test app does this and the things it does show up.

File: platform_windows.cs (hunk 1 above)
Comment (drew-richardson) [reply platform_windows.cs:67]: OK, then it's in the wrong place right now, but I haven't added any focus logic yet, but this way the methods are called before the user clicks on the window. Then when I add proper focus/unfocus handling, I'll move it there.

File: platform_windows.go (hunk 4 above)
Comment (drew-richardson) [reply platform_windows.go:83]: It doesn't. This is part of the hack I'm currently doing with the `RootWindow`. I have some logic in place to clone the menu across all the windows, but I don't think it's quite working properly yet.

**Review by richardwilkes — APPROVED** [review]

**Comment by richardwilkes** [comment]
Good start. Will merge and we can work on improving this.

**Pull request merged**